### PR TITLE
update wasmex,wasmer,rustler,& rust dependencies

### PR DESCRIPTION
**Pull request by stevelr**
Updated:

- [wasmex 0.7.0 from 0.5.0](https://github.com/tessi/wasmex/blob/main/CHANGELOG.md), which updates wasmer to 2.1.1
- [rustler 0.24 from 0.23](https://github.com/rusterlium/rustler/blob/master/CHANGELOG.md)
- updated other rust dependencies via 'cargo update'

Signed-off-by: Steve <steve@cosmonic.com>

host_core tests pass. Creating as draft pr to allow additional testing

### CODE REVIEW

**Comment by stevelr** [comment]
also updated build runner from ubuntu 18.04 to 20.04.

Apparently, this version of ubuntu includes a mono dev environment that automatically runs a monodoc http server on port 8084, so had to change the echo/httpserver test to use a different port number.
The monodoc service can be stopped with 
`sudo systemctl stop mono-xsp4.service` but it was simpler to change the port than the workflow script.

**Comment by stevelr** [comment]
I don't know how to get rid of those expected checks for 1.12, since 1.12 isn't in the matrix anymore.

**Comment by thomastaylor312** [comment]
@stevelr We can remove those from branch protections once we merge this

**Review by thomastaylor312 — COMMENTED** [review]
One minor change: Also, do we need to update the elixir version of the lattice controller as well so they play nice together?

File: host_core/mix.exs [hunk 1]
```diff
@@ -30,11 +30,13 @@ defmodule HostCore.MixProject do
...
   defp deps do
     [
-      {:wasmex, "~> 0.5.0"},
+      {:wasmex, "~> 0.7.0"},
       {:msgpax, "~> 2.3"},
-      {:rustler, "~> 0.22.0"},
+      {:rustler, "~> 0.24.0"},
       {:jason, "~> 1.2.2"},
       {:gnat, "~> 1.2"},
+      # erlavro isn't used, but this version upgrades dependency of cloudevents 0.4.0 to use rebar3
```
Comment (autodidaddict) [inline host_core/mix.exs:38]: This feels a little off. We're including a dependency that we don't need in order to upgrade the way another dependency works?
surely there's a better way to do this?

**Review by autodidaddict — COMMENTED** [review]
Just a question about adding a dependency we don't need...

**Comment by autodidaddict** [comment]
> One minor change: Also, do we need to update the elixir version of the lattice controller as well so they play nice together?

If, by `lattice controller`, you mean `lattice observer`, then it doesn't need to change yet because there's no direct dependency. At the moment, the only coupling between the lattice observer and the OTP host is the payload contract on the `wasmbus.evt.*` topic.

**Comment by autodidaddict** [comment]
This fixes #331

File: host_core/mix.exs (hunk 1 above)
Comment (autodidaddict) [reply host_core/mix.exs:38]: There's got to be a better way to fix this, but for now let's just move forward.

**Review by autodidaddict — APPROVED** [review]

**Pull request merged**